tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -190,6 +192,7 @@ void test_single_shot_alarm_instance(const char *dev_name, bool set_top)
 
 		zassert_equal(0, err, "Counter failed to set top value\n");
 
+		alarm_cfg.ticks = ticks + 1;
```

Comment:
This is not directly related but why is the number of ticks decreased by 100 in line [200](https://github.com/zephyrproject-rtos/zephyr/pull/13305/files#diff-cbfda9aaae5d9069931b177bfff91d59R200)? This happens only when the top value is modified. Is this intentional? Maybe a word of comment could help there?